project-management/src/main/java/life/qbic/projectmanagement/application/api/AsyncProjectService.java

```diff
@@ -1469,15 +1481,9 @@ record ValidationRequest(String projectId, ValidationRequestBody requestBody,
                            String requestId) implements CacheableRequest {
 
     public ValidationRequest {
-      if (projectId == null) {
-        throw new IllegalArgumentException("Project ID cannot be null");
-      }
-      if (projectId.isBlank()) {
-        throw new IllegalArgumentException("Project ID cannot be blank");
-      }
-      if (requestId == null || requestId.isBlank()) {
-        requestId = UUID.randomUUID().toString();
-      }
+      requireNonNull(projectId);
+      requestId =
+          isNull(requestId) || requestId.isBlank() ? UUID.randomUUID().toString() : requestId;
```

Comment:
i think this is a open door for nasty side effects. The client either provides the ID or must explicitly call the constructor method for creating an ID automatically.